pages/index.tsx

```diff
@@ -0,0 +1,169 @@
+import Head from "next/head";
+import styles from "../styles/Home.module.css";
+import React, { useEffect, useRef, useState } from "react";
+import fetch from "node-fetch";
+import * as jwt from "jsonwebtoken";
+import {
+  CLIENT_ID,
+  HOSTED_UI_LOGIN_URL,
+  HOSTED_UI_TOKEN_URL,
+  REDIRECT_URI,
+} from "./api/constants";
+import extractUrlParameter from "./api/extractUrlParameter";
+import generateSaml from "./api/generateSaml";
+import {
+  DecodedIdToken,
+  IdDetails,
+  SsoDetails,
+  TokenResponse,
+} from "./api/types";
+import CallCenterPicker from "../components/CallCenterPicker";
+import SelfSubmittingSsoForm from "../components/SelfSubmittingSsoForm";
+
+export default function Home() {
```

Comment:
So this looks like a server component? I'm still learning the Next intricacies, any reason for this being a server vs. client

Reply:
Not sure actually. Server components are supported as of React 18 which this code is using, but I intentionally built on Next v12 instead of v13 because v13 is where the server/client split becomes really real... and I haven't learned enough about that yet to be good at it.